relengapi/blueprints/archiver/__init__.py

```
@@ -65,6 +121,8 @@ def get_hgmo_archive(repo, rev, subdir=None, suffix='tar.gz', preferred_region=N
     :param suffix: the archive extension type. defaulted to tar.gz
     :param preferred_region: the preferred s3 region to use
     """
+    # allow for the short hash and full hash to be passed
+    rev = rev[0:12]
```

It seems like this should be a client-side requirement, rather than silently discarding parts of the request.  Can you land this as-is, but also update the client to always truncate revs to 12 bytes?  Once that'd deployed, we can update this service to fail with 400 if `len(rev) != 12`.